include/nanorange/algorithm/stable_partition.hpp

```diff
@@ -95,37 +99,40 @@ struct stable_partition_fn {
         }
 
         const I first_false = (m1 == first) ? first :
-                impl_unbuffered(first, m1, len_half, pred, proj);
+                impl_unbuffered(first, m1, len_half, pred, proj).end();
 
         m1 = middle;
         len_half = dist - half;
 
         while (nano::invoke(pred, nano::invoke(proj, *m1))) {
             if (++m1 == last) {
-                return nano::rotate(first_false, middle, ++last).begin();
+                auto rot = nano::rotate(first_false, middle, ++last);
+                return {first_copy, rot.begin()};
             }
         }
 
-        const I last_false = impl_unbuffered(m1, last, len_half, pred, proj);
+        const I last_false = impl_unbuffered(m1, last, len_half, pred, proj).end();
 
-        return nano::rotate(first_false, middle, last_false).begin();
+        auto rot = nano::rotate(first_false, middle, last_false);
+        return {first_copy, rot.begin()};
     }
 
     template <typename I, typename Pred, typename Proj>
-    static I impl(I first, I last, Pred& pred, Proj& proj)
+    static subrange<I> impl(I first, I last, Pred& pred, Proj& proj)
     {
+        I first_copy = first;
         // Find the first non-true value
         first = nano::find_if_not(std::move(first), last, std::ref(pred), std::ref(proj));
```

Comment:
Again, `auto iter = find_if_not(first, last, ref(pref), ref(proj))` looks cleaner to me

Reply:
Fixed.